{
    "comments": [
        {
            "author": null,
            "body": "Fixed in trunk and proposed for 6.0.x",
            "date": "20100525T10:30:40",
            "id": 0
        },
        {
            "author": null,
            "body": "Created attachment 25534\n2010-06-07_tc55_bug49196.patch\n\nPatch for 5.5. It is backport of r948057 from trunk.",
            "date": "20100606T16:07:12",
            "id": 1
        },
        {
            "author": null,
            "body": "The fix was applied to 6.0 in r951941 and will be in 6.0.27.\n\nProposed for backport to 5.5, so I am changing Product and Version to track this issue further.",
            "date": "20100606T16:15:26",
            "id": 2
        },
        {
            "author": null,
            "body": "This has been fixed in 5.5.x and will be included in 5.5.30 onwards.",
            "date": "20100622T05:01:05",
            "id": 3
        }
    ],
    "component": "Catalina",
    "description": "This bug is present in some older versions also, but I haven't had time until now to file the report. Here it is:\n\nI've got\n\n  <error-page>\n    <error-code>403</error-code>\n    <location>/free/login/denied.jsp</location>\n  </error-page>\n\nin web.xml and call pageContext.getErrorData() in denied.jsp.\n\nTHIS WORKS FINE!\n\nWhat does not work is if the user is HTTP-redirected to denied.jsp or simply calls it by entering the URL in his browser. In this case the JSP throws\n\njava.lang.NullPointerException\n        at javax.servlet.jsp.PageContext.getErrorData(PageContext.java:515)\n        at org.apache.jsp.free.login.denied_jsp._jspService(denied_jsp.java:389)\n        at org.apache.jasper.runtime.HttpJspBase.service(HttpJspBase.java:70)\n        at javax.servlet.http.HttpServlet.service(HttpServlet.java:717)\n        at org.apache.jasper.servlet.JspServletWrapper.service(JspServletWrapper.java:377)\n        at org.apache.jasper.servlet.JspServlet.serviceJspFile(JspServlet.java:313)\n        at org.apache.jasper.servlet.JspServlet.service(JspServlet.java:260)\n        at javax.servlet.http.HttpServlet.service(HttpServlet.java:717)\n        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:290)\n        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206)\n        at com.folgmann.filters.AuthFilter.doFilter(AuthFilter.java:446)\n        at com.folgmann.filters.HttpFilterSupport.doFilter(HttpFilterSupport.java:41)\n        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:235)\n        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206)\n        at com.folgmann.filters.ValidatingThreadPoolFilter.doFilter(ValidatingThreadPoolFilter.java:138)\n        at com.folgmann.filters.HttpFilterSupport.doFilter(HttpFilterSupport.java:41)\n        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:235)\n        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206)\n        at com.folgmann.filters.CharsetFilter.doFilter(CharsetFilter.java:114)\n        at com.folgmann.filters.HttpFilterSupport.doFilter(HttpFilterSupport.java:41)\n        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:235)\n        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206)\n        at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:233)\n        at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:191)\n        at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:465)\n        at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:127)\n        at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:102)\n        at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:109)\n        at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:298)\n        at org.apache.jk.server.JkCoyoteHandler.invoke(JkCoyoteHandler.java:190)\n        at org.apache.jk.common.HandlerRequest.invoke(HandlerRequest.java:291)\n        at org.apache.jk.common.ChannelSocket.invoke(ChannelSocket.java:769)\n        at org.apache.jk.common.ChannelSocket.processConnection(ChannelSocket.java:698)\n        at org.apache.jk.common.ChannelSocket$SocketConnection.runIt(ChannelSocket.java:891)\n        at org.apache.tomcat.util.threads.ThreadPool$ControlRunnable.run(ThreadPool.java:690)\n        at java.lang.Thread.run(Thread.java:619)\n\nI'm sure that this wasn't intended by the JSP spec. I suggest that pageContext.getErrorData() returns null instead when an error page is not called like an error page should be called.\n\nTnx,\n     boris",
    "hasPatch": true,
    "hasScreenshot": false,
    "id": "49196",
    "issuetypeClassified": "BUG",
    "issuetypeTracker": "BUG",
    "priority": "P2 normal",
    "product": "Tomcat 5",
    "project": "TOMCAT",
    "summary": "NPE in PageContext.getErrorData()",
    "systemSpecification": true,
    "version": "5.5.28"
}